platform-tests/src/acceptance/rds_broker_test.go

```diff
@@ -124,8 +133,108 @@ var _ = Describe("RDS broker", func() {
 			Expect(resp.StatusCode).To(Equal(200), "Got %d response testing multi-user permissions. Response body:\n%s\n", resp.StatusCode, string(body))
 		})
 	})
+
+	Context("creating a database instance with custom parameters", func() {
+
+		var (
+			dbInstanceName  string
+			rdsInstanceName string
+		)
+
+		BeforeEach(func() {
+			dbInstanceName = generator.PrefixedRandomName("test-db-")
+			Expect(cf.Cf("create-service", serviceName, testPlanName, dbInstanceName, "-c", `{"skip_final_snapshot": "true"}`).Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+
+			pollForRDSCreationCompletion(dbInstanceName)
+
+			rdsInstanceName = getRDSInstanceName(dbInstanceName)
+			fmt.Fprintf(GinkgoWriter, "Created RDS instance: %s\n", rdsInstanceName)
+		})
+
+		It("should not create a final snapshot when `skip_final_snapshot` is set to true", func() {
+			Expect(cf.Cf("delete-service", dbInstanceName, "-f").Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+
+			// Poll until destruction is complete, otherwise the org cleanup fails.
+			pollForRDSDeletionCompletion(dbInstanceName)
+
+			rdsClient, err := NewRDSClient(region)
+			Expect(err).NotTo(HaveOccurred())
+			snapshots, err := rdsClient.GetDBFinalSnapshots(rdsInstanceName)
+			fmt.Fprintf(GinkgoWriter, "Final snapshots for %s:\n", rdsInstanceName)
+			fmt.Fprint(GinkgoWriter, snapshots)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(ContainSubstring("DBSnapshotNotFound"))
+
+			snapshotDeletionOutput, err := rdsClient.deleteDBFinalSnapshot(rdsInstanceName)
+			fmt.Fprintf(GinkgoWriter, "Snapshot deletion output for %s:\n", rdsInstanceName)
+			fmt.Fprint(GinkgoWriter, snapshotDeletionOutput)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should create a final snapshot if we set `skip_final_snapshot` back to false", func() {
+			Expect(cf.Cf("update-service", dbInstanceName, "-c", `{"skip_final_snapshot": "false"}`).Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+			pollForRDSUpdateCompletion(dbInstanceName)
+			Expect(cf.Cf("delete-service", dbInstanceName, "-f").Wait(DEFAULT_TIMEOUT)).To(Exit(0))
+
+			// Poll until destruction is complete, otherwise the org cleanup fails.
+			pollForRDSDeletionCompletion(dbInstanceName)
+
+			rdsClient, err := NewRDSClient(region)
+			Expect(err).NotTo(HaveOccurred())
+			snapshots, err := rdsClient.GetDBFinalSnapshots(rdsInstanceName)
+			fmt.Fprintf(GinkgoWriter, "Final snapshots for %s:\n", rdsInstanceName)
+			fmt.Fprint(GinkgoWriter, snapshots)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(snapshots).Should(ContainSubstring(rdsInstanceName))
+
+			snapshotDeletionOutput, err := rdsClient.deleteDBFinalSnapshot(rdsInstanceName)
+			fmt.Fprintf(GinkgoWriter, "Snapshot deletion output for %s:\n", rdsInstanceName)
+			fmt.Fprint(GinkgoWriter, snapshotDeletionOutput)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
 })
 
```

Comment:
I can't help but feel that the following three functions could be refactored into one, however in the interest of getting The Neverending Story merged I think we should ignore this.